src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -1731,8 +1552,10 @@ wordsBy predicate f (Stream step state) = Stream (stepOuter f) (Just state)
                 else do
                     acc <- initial
                     acc' <- fstep acc x
-                    go SPEC s acc'
-
+                    case acc' of
+                        FL.Partial sres -> go SPEC s sres
+                        FL.Done bres -> go1 SPEC s bres
```

Comment:
> if the fold has terminated it means we have reached a word boundary

This is not true, but since we give the fold control to dictate the stream output I need to change the behaviour as if we have reached a word boundary.